napari/_qt/widgets/qt_dims.py

```diff
@@ -339,16 +338,11 @@ def _set_frame(self, axis, frame):
         the canvas can draw, this will drop the intermediate frames, keeping
         the effective frame rate constant even if the canvas cannot keep up.
         """
-        if self._play_ready:
+        if self.dims._play_ready:
             # disable additional point advance requests until this one draws
-            self._play_ready = False
+            self.dims._play_ready = False
             self.dims.set_current_step(axis, frame)
 
-    def enable_play(self, *args):
-        # this is mostly here to connect to the main SceneCanvas.events.draw
-        # event in the qt_viewer
-        self._play_ready = True
-
```

Comment:
In this case, the connect event was in the _create_canvas method of the QtViewer class. As stated it only is there to connect to the draw event that is now moved to the VispyCanvas. From what I could tell the code was related to allow for animations to be played properly. Conceptually it seemed to make sense to me that this should actually be a property of the Napari dims EventedModel. It is not something QT specific. Does this make sense?

Reply:
Thanks for the explanation. I guess playback animation (of the dimension sliders) is not necessarily Qt specific (i.e. you could imagine different implementations), but I do think it's viewer, and not viewer/dims model, specific. 

I.e. playback animation of the dimension sliders is something that the GUI drives, which causes changes to the viewer/dims model (i.e. specifically `Dims.current_step`). This is reflected in the current implementation of `QtDims` where `_play_ready` is used to prevent too many updates to `Dims.current_step`.

There could be some value is allow model-based playback (e.g. regularly timed updates to `Dims.current_step`), but we'd need to do a lot more to support that, so just adding `Dims.play_ready` expands napari's public API without any obvious immediate benefit.

Therefore, if we can leave this where it is or find a new home for it outside of napari's public API, I think that would be best.

Reply:
I think I understand what you mean. However, having true multi-canvas in mind at a later stage where one could be dealing with different Dims models would it still be viewer specific. Also, is this something that we should already think of now? If I am thinking about this right it could change the user API again at a later stage. That is if I understand its purpose correctly. It would be great to hear your thoughts on this.

Reply:
To me, it's less about viewer (or viewermodel) vs. dims, and more about view vs. model (in the MVC design pattern), GUI vs. API, or implementation detail vs. public API. Playback animation is a GUI feature that is not currently exposed in the public API. My main concern here is that the feature is creeping into the public API, but only partially and not in a useful way.

If you can't find a better place for the code than `Dims` right now, I'd prefer to make the method private even though it is accessed from outside that class rather than expanding the public API. It's fine to think about now, but we should probably have a solid idea about how the public API will look before we start making changes to it on `main`.

I can also try to take a look to see if I can move the code elsewhere, but can't do that until next week.

Reply:
I will have another look during the weekend

Reply:
I have had another look and came up with [f2dd656](https://github.com/napari/napari/pull/5432/commits/f2dd656720fec8f551083105cbdddaec1d1596b0). Enable play exists to just connect a draw event from the Vispy SceneCanvas, thus I believe the VispyCanvas should be the controller in this case. Thus I moved the method to VispyCanvas. The _play_ready attribute stays private within the Napari Dims model and Qt can interact with it by directly accessing / setting the attribute when required.
@andy-sweet @brisvag What do you think?

Reply:
This is good enough for me approve this PR, since there's nothing new in the napari public API.

In the long term, I think that if we have multiple instances of `VispyCanvas`, then we might have one instance of `QtDims` per instance of `VispyCanvas` because each canvas displays dimensions, so this might end up getting moved back to `QtDims` from `Dims`. But right now, it's hard to speculate.